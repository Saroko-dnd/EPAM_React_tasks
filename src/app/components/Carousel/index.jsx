import React from 'react';
import PropTypes from 'prop-types';
import { Carousel } from 'react-responsive-carousel';

import './scss/carousel.scss';

class CustomCarousel extends React.Component {
    shouldComponentUpdate() {
        return this.props.elements.length === 0;
    }

    render() {
        return (
            <div className="carousel-wrapper">
                <Carousel
                    render={() => <h5>dfdfdf</h5>}
                    onChange={this.props.onChange}
                    className="d-block"
                    showThumbs={false}
                    width="100%"
                >
                    {this.props.elements}
                </Carousel>
            </div>
        );
    }
}

CustomCarousel.propTypes = {
    elements: PropTypes.arrayOf(PropTypes.element),
    onChange: PropTypes.func.isRequired,
};

CustomCarousel.defaultProps = {
    elements: [],
};

export default CustomCarousel;
