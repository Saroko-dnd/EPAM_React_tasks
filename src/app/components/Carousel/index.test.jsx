import Carousel from '.';

describe('render a Carousel', () => {
    it('should render Carousel correctly', () => {
        const wrapper = shallow(<Carousel title="rrr" onChange={() => {}} />);
        expect(wrapper).toMatchSnapshot();
    });
});
