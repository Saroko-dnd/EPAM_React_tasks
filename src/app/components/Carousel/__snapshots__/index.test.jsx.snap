// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Carousel should render Carousel correctly 1`] = `
<div
  className="carousel-wrapper"
>
  <Carousel
    autoPlay={false}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    className="d-block"
    dynamicHeight={false}
    emulateTouch={false}
    infiniteLoop={false}
    interval={3000}
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    render={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={false}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="100%"
  />
</div>
`;
